services/src/messaging/src/main/java/org/openkilda/messaging/payload/flow/FlowPayloadToFlowConverter.java

```diff
@@ -70,24 +73,41 @@ public static FlowPayload buildFlowPayloadByFlow(Flow flow) {
     }
 
     /**
-     * Builds list of {@link FlowPayload} instances by list of {@link Flow} instance.
+     * Builds {@link FlowPayload} instance by {@link ImmutablePair} instance.
      *
-     * @param flows list of {@link Flow} instance
-     * @return list of {@link FlowPayload} instance
+     * @param flow {@link BidirectionalFlow} the bidirectional flow with paths
+     * @return {@link FlowPayload} instance
      */
-    public static List<FlowPayload> buildFlowsPayloadByFlows(List<Flow> flows) {
-        return flows.stream().map(FlowPayloadToFlowConverter::buildFlowPayloadByFlow).collect(Collectors.toList());
+    public static FlowPathPayload buildFlowPathPayloadByBidirectionalFlow(BidirectionalFlow flow) {
```

Comment:
Nmae chunk `ByBidirectionalFlow` is excess. You can create several method `buildFlowPathPayload` with different set of arguments. It is much more readable.